src/BuildToolchains.jl

```diff
@@ -179,9 +197,9 @@ function toolchain_file(bt::Meson, p::AbstractPlatform, envs::Dict{String,String
     fortran = '/opt/bin/$(target)/$(aatarget)-$(fortran_compiler(bt))'
     objc = '/opt/bin/$(target)/$(aatarget)-cc'
     ar = '/opt/bin/$(target)/$(aatarget)-ar'
-    ld = '$(linker_string(p, clang_use_lld))'
-    cpp_ld = '$(linker_string(p, clang_use_lld))'
-    c_ld = '$(linker_string(p, clang_use_lld))'
+    ld = '$(linker_string(bt, p, clang_use_lld))'
+    cpp_ld = '$(linker_string(bt, p, clang_use_lld))'
+    c_ld = '$(linker_string(bt, p, clang_use_lld))'
```

Comment:
I thought here you needed the basename of the linker?

Reply:
It's not even basename. GCC wants an actual name. While clang understands a full path. So this dispatches on the compiler now